include/mxnet/ndarray.h

```diff
@@ -1057,6 +1119,41 @@ class NDArray {
     bool IsDefault() const;
 #endif
 
+    /// \brief Cast storage type and auxiliary index to string,
+    ///        used for recording storage allocations.
+    inline std::string STypeToString() const {
+      if (storage_type == kUndefinedStorage) {
+        return "undef";
+      } else if (storage_type == kRowSparseStorage) {
+        return "row_sparse";
+      } else if (storage_type == kCSRStorage) {
+        return "csr";
+      } else {
+        return "unknown";
+      }
+    }
```

Comment:
@eric-haibin-lin do you have suggestions on where to put this?

Reply:
@eric-haibin-lin Sorry I did not notice that. Thanks.